returnn/torch/engine.py

```diff
@@ -193,13 +250,29 @@ def train_epoch(self):
                 {name: float(_to_raw(loss.get_inv_norm_factor())) for name, loss in train_ctx.losses.items()}
             )
 
-            if self._grad_scaler is not None:
-                self._grad_scaler.scale(total_loss).backward()
-                self._grad_scaler.step(self._updater.get_optimizer())
-                self._grad_scaler.update()
+            if self._use_torch_distributed and (step_idx % self._accum_grad_multiple_step) != (
```

Comment:
Do not put the `if` statement here. Do not repeat the same code below twice. Rather, Just have the code once, but have the context manager optionally, and use `nullcontext` otherwise.